scripts/core/MultiSelectHoc.tsx

```diff
@@ -86,7 +98,7 @@ export class MultiSelectHoc extends React.PureComponent<IProps, IState> {
     handleContentChanges(resource: string, id: string) {
         // Unselect items that no longer match the query.
 
-        if (ARTICLE_RELATED_RESOURCE_NAMES.includes(resource) && this.state.selected.has(id)) {
+        if (this.props.resourceNames.includes(resource) && this.state.selected.has(id)) {
```

Comment:
imo it should be using `getId` when checking `state.selected`, shouldn't it?

Reply:
It's a long story :upside_down_face:

Ideally, I should not need `getId` at all, and would use the `_id` field that is present for every resource.

Since that `_id` field is not unique for published items as we already discussed, it doesn't work here. I'm trying to work around it using `getId` which does the job in most cases.

Here's it's messy, because I assume that IDs received from a websocket notification are the same(have version appended like `generateTrackByIdentifier` does) as `getId` would produce if I had a full object(which I don't).

If IDs from websocket notifications don't have version appended, then I'm a bit out of options. I may be able to restructure `handleContentChanges` so it's passed as a prop from a parent component and could fetch all the items with that duplicate id from the websocket notification and then use `getId`, but that feels like a sandwich of hacks.

The way it's implemented now would only misbehave when users select published items for bulk actions and those published items would be modified by someone else and would no longer match the query that is used to render the list.

It'd be good to fix the root cause - that duplicate ID issue so I could get rid of off `getId` workaround. I ended up not using the fix you did [here](https://github.com/superdesk/superdesk-core/pull/2112), because I realized that even if I add that query parameter, it would not work. It would work for brand new code, but I wanted to use existing components, that have their own child components that don't know about the query parameter and wouldn't work in both scenarios.

Reply:
I see, we should plan for the fix without any param probably for some future version then.